drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -2222,6 +2222,9 @@ static int bcm2835_codec_s_ctrl(struct v4l2_ctrl *ctrl)
 		container_of(ctrl->handler, struct bcm2835_codec_ctx, hdl);
 	int ret = 0;
 
+	if (ctrl->flags & V4L2_CTRL_FLAG_READ_ONLY)
+		return 0;
```

Comment:
Good job defining controls as read-only and exiting early from s_ctrl on read-only controls.